The error looks odd:
```
16:55:07  #18 DONE 0.3s
16:55:07  docker tag cilium/hubble-relay:5d82d5084a83d2bbe06c7cc87eacd9fb7c7c90f8 cilium/hubble-relay:5d82d5084a83d2bbe06c7cc87eacd9fb7c7c90f8-amd64
16:55:07  Push like this when ready:
16:55:07  docker push cilium/hubble-relay:5d82d5084a83d2bbe06c7cc87eacd9fb7c7c90f8-amd64
16:55:07  make docker-operator-image
16:55:07  make[1]: Entering directory '/home/jenkins/workspace/Cilium-PR-K8s-GKE@2/src/github.com/cilium/cilium'
16:55:07  HEAD detached at 5d82d5084a
16:55:07  Untracked files:
16:55:07    (use "git add <file>..." to include in what will be committed)
16:55:07  
16:55:07  	test/gke/registry-adder.yaml
16:55:07  
16:55:07  nothing added to commit but untracked files present (use "git add" to track)
16:55:07  These changes will not be included in build, aborting. Define IGNORE_GIT_STATUS to build anyway.
16:55:07  Makefile.buildkit:73: recipe for target 'check-status' failed
16:55:07  make[1]: *** [check-status] Error 1
16:55:07  make[1]: Leaving directory '/home/jenkins/workspace/Cilium-PR-K8s-GKE@2/src/github.com/cilium/cilium'
16:55:07  Makefile.docker:16: recipe for target 'docker-image' failed
16:55:07  make: *** [docker-image] Error 2
```

--- a/.gitignore
+++ b/.gitignore
@@ -74,6 +74,7 @@ test/gke/cluster-uri
 test/gke/cluster-version
 test/gke/gke-kubeconfig
 test/gke/resize-kubeconfig
+test/gke/registry-adder.yaml
 
 # Emacs backup files
 *~

--- a/test/gke/select-cluster.sh
+++ b/test/gke/select-cluster.sh
@@ -20,6 +20,13 @@ while [ $locked -ne 0 ]; do
     echo "selecting random cluster"
     cluster_uri="$(gcloud container clusters list --project "${project}" --filter="name ~ ^cilium-ci-" --uri | sort -R | head -n 1)"
 
+	echo "checking whether cluster ${cluster_uri} has any node pools"
+	node_pools=$(gcloud container node-pools list --project "${project}" --region "${region}" --cluster "${cluster_uri}" --format="value(name)")
+	if [ -z "${node_pools}" ]; then
+		echo "no nodepools found, selecting another cluster"
+		continue
+	fi
+
     echo "getting kubeconfig for ${cluster_uri} (will store in ${KUBECONFIG})"
     gcloud container clusters get-credentials --project "${project}" --region "${region}" "${cluster_uri}"
 

